projects/js-packages/publicize-components/src/hooks/use-image-generator-config/index.js

```diff
@@ -23,35 +25,41 @@ import { useCallback } from '@wordpress/element';
 export default function useImageGeneratorConfig() {
 	const { editPost } = useDispatch( editorStore );
 
-	const { postSettings, currentOptions, isPostPublished } = useSelect( select => ( {
-		postSettings: select( 'jetpack/publicize' ).getImageGeneratorPostSettings(),
-		currentOptions: select( 'jetpack/publicize' ).getJetpackSocialOptions(),
+	const { postSettings, currentOptions } = useSelect( select => ( {
+		postSettings: select( PUBLICIZE_STORE_ID ).getImageGeneratorPostSettings(),
+		currentOptions: select( PUBLICIZE_STORE_ID ).getJetpackSocialOptions(),
+	} ) );
+
+	const { isPostPublished } = useSelect( select => ( {
 		isPostPublished: select( editorStore ).isCurrentPostPublished(),
 	} ) );
 
 	const updateSettings = useCallback(
-		settings => {
+		( key, value ) => {
+			const settings = { ...postSettings, [ key ]: value };
 			editPost( {
 				meta: {
 					jetpack_social_options: { ...currentOptions, image_generator_settings: settings },
 				},
 			} );
 		},
-		[ currentOptions, editPost ]
+		[ currentOptions, editPost, postSettings ]
 	);
 
-	const updateSetting = ( setting, value ) =>
-		updateSettings( { ...postSettings, [ setting ]: value } );
-	const getSetting = setting => currentOptions?.image_generator_settings?.[ setting ] ?? null;
+	const getCurrentSettings = sigSettings => {
+		return {
+			isEnabled: sigSettings?.enabled ?? ! isPostPublished,
+			customText: sigSettings?.custom_text ?? null,
+			imageType: sigSettings?.image_type ?? null,
+			imageId: sigSettings?.image_id ?? null,
+		};
+	};
```

Comment:
Real nitpick, but I think this could be:
```suggestion
	const getCurrentSettings = sigSettings => ( {
		isEnabled: sigSettings?.enabled ?? ! isPostPublished,
		customText: sigSettings?.custom_text ?? null,
		imageType: sigSettings?.image_type ?? null,
		imageId: sigSettings?.image_id ?? null,
	) };
```